### ISSUE TITLE
wait-on-timeout Value Doesn't Appear to Be Used

### ISSUE THREAD

**Issue by chriswsh** [cypress-io/github-action#313]
With these settings:

```
uses: cypress-io/github-action@v2
with:
  start: npm start
  wait-on: "http://localhost:3000"
  wait-on-timeout: 300
```

I get these logs in Github actions:

```
start server "npm start command "npm start"
current working directory "/home/runner/work/badger-frontend/badger-frontend"
waiting on "http://localhost:3000" with timeout of 60 seconds
```

I'd expect that the wait time for `http://localhost:3000` would be the `wait-on-timeout` value.

Possibly related to the behavior in https://github.com/cypress-io/github-action/issues/286

**Comment by jrparish** [comment]
Interesting, I was having problems _until_ I set this value. My settings are very similar to yours and are working: 

![image](https://user-images.githubusercontent.com/5173987/112668997-288e0280-8e2d-11eb-864a-2e5b1f60d8bc.png)
![image](https://user-images.githubusercontent.com/5173987/112668955-1b711380-8e2d-11eb-9cda-d152643a94b4.png)

**Comment by bahmutov** [comment]
Please provide the relevant portion of the debug logs for us to see if anything is going wrong

https://github.com/cypress-io/github-action#debugging

In https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-wait-on.yml we have multiple test examples for waiting for the server to start, so your case is probably something very different that we did not account for. Then we can re-open this issue.

**Comment by chriswsh** [comment]
@bahmutov Thanks for checking back on this.

I'm unable to reproduce the issue.

The only other thing I can think of is that, when the action was failing consistently, the action to run the Cypress tests was triggered on push. That sometimes resulted in several runs on several branches happening simultaneously, and those attempts errored out with the message listed in #286.

At first, waiting 10 to 15 minutes, then rerunning the action would resolve. Then the runs started to fail 100% of the time.

In an attempt to resolve that, I added the wait-on-timeout value to the action config. I consistently got the strange behavior mentioned here, along with the failure message from #286.

By that point, we probably had 1,000% or more of our monthly test runs via the free Cypress Dashboard plan used.

I spent most of today spamming test runs, up to 174% of the free allotment. No errors yet, and the logs show that the action is indeed waiting for the server to start for the correct number of seconds.

We've since moved the Cypress testing action to trigger when a release is cut, rather than every push. If we run into this again, I'll get the logs and post them here.
